lib/presentation/custom_widgets/fab_button/expandable_fab.dart

```diff
@@ -0,0 +1,143 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+
+class ExpandableFab extends HookWidget {
+  const ExpandableFab({
+    Key? key,
+    this.initialOpen = false,
+    required this.children,
+  }) : super(key: key);
+
+  final bool initialOpen;
+
+  final List<Widget> children;
+
+  @override
+  Widget build(BuildContext context) {
+    final isOpen = useState(initialOpen);
+    final animationController = useAnimationController(
+      initialValue: isOpen.value ? 1 : 0,
+      duration: const Duration(milliseconds: 250),
+    );
+
+    void _toggle() {
+      isOpen.value = !isOpen.value;
+      if (isOpen.value) {
+        animationController.forward();
+      } else {
+        animationController.reverse();
+      }
+    }
+
+    return SizedBox.expand(
+      child: Stack(
+        alignment: Alignment.bottomRight,
+        clipBehavior: Clip.none,
+        children: [
+          _buildTapToCloseFab(context, onPressed: _toggle),
+          ..._buildExpandingActionButtons(animationController),
+          _buildTapToOpenFab(isOpen.value, onPressed: _toggle),
+        ],
+      ),
+    );
+  }
+
+  List<Widget> _buildExpandingActionButtons(AnimationController animation) {
+    final children = <Widget>[];
+
+    this.children.asMap().forEach((i, value) {
+      children.add(
+        _ExpandingActionButton(
+          index: i,
+          progress: CurvedAnimation(
+            curve: Curves.fastOutSlowIn,
+            reverseCurve: Curves.easeOutQuad,
+            parent: animation,
+          ),
+          child: this.children[i],
+        ),
+      );
+    });
+    return children;
```

Comment:
**suggestion**: There is a little cleaner way to do this...

```suggestion
    return children.asMap().keys.map((index) {
        return _ExpandingActionButton(
          index: index,
          progress: CurvedAnimation(
            curve: Curves.fastOutSlowIn,
            reverseCurve: Curves.easeOutQuad,
            parent: animation,
          ),
          child: children[index],
        ),
      );
    }).toList();
```

Reply:
Cleaner indeed!